### PR TITLE
CPR-486-memory-error-fix

### PR DESCRIPTION
**Pull request by JunaedAhmedMOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt [hunk 1]
```diff
@@ -49,56 +49,63 @@ class CourtEventProcessor(
...
-    val uniqueDefendants = commonPlatformHearingEvent.hearing.prosecutionCases
-      .flatMap { it.defendants }
-      .filterNot { it.isYouth }
-      .distinctBy {
-        it.personDefendant?.personDetails?.firstName +
-          it.personDefendant?.personDetails?.lastName +
-          it.personDefendant?.personDetails?.dateOfBirth +
-          it.pncId +
-          it.cro
+    run {
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt:52]: What effect is this having?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt [hunk 2]
```diff
@@ -49,56 +49,63 @@ class CourtEventProcessor(
...
+            EventKeys.MESSAGE_ID to sqsMessage.messageId,
+            EventKeys.SOURCE_SYSTEM to SourceSystemType.COMMON_PLATFORM.name,
+          ),
+        )
+        createUpdateService.processMessage(person) {
+          person.defendantId?.let {
+            personRepository.findByDefendantId(it)
+          }
+        }
+        person.defendantId = null
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt:82]: What is the reason for this? this wont effect any thing as it is set after processing

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt [hunk 3]
```diff
@@ -49,56 +49,63 @@ class CourtEventProcessor(
...
-        EventKeys.EVENT_TYPE to LIBRA_COURT_CASE.name,
-        EventKeys.MESSAGE_ID to sqsMessage.messageId,
-        EventKeys.SOURCE_SYSTEM to SourceSystemType.LIBRA.name,
-      ),
-    )
-    createUpdateService.processMessage(person) {
-      val personEntity = personService.searchBySourceSystem(person)
-      person.defendantId = personEntity?.defendantId ?: UUID.randomUUID().toString()
-      return@processMessage personEntity
+      person.defendantId = null
```
Comment (H-Iwanejko) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt:108]: as above

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt [hunk 4]
```diff
@@ -45,67 +44,60 @@ class CourtEventProcessor(
...
-          ),
-        )
-        createUpdateService.processMessage(person) {
-          person.defendantId?.let {
-            personRepository.findByDefendantId(it)
-          }
-        }
-        person.defendantId = null
+    val jsonParser = objectMapper.factory.createParser(sqsMessage.message)
+    val commonPlatformHearingEvent = jsonParser.readValueAs(CommonPlatformHearingEvent::class.java)
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt:48]: What does this do that the existing code doesn't?

**Pull request closed without merging**